pkg/tun/tcp/handler.go

```diff
@@ -313,14 +313,13 @@ func (h *handler) sendSyn(ctx context.Context, ackNumber uint32, ack bool) {
 // writeToTunLoop sends the packages read from the fromMgr channel to the TUN device
 func (h *handler) writeToTunLoop(ctx context.Context) {
 	defer close(h.readyToFin)
-	ticker := time.NewTicker(100 * time.Millisecond)
 	for {
 		var cm connpool.Message
 		select {
 		case <-ctx.Done():
 			return
 		case cm = <-h.fromMgr:
-		case <-ticker.C:
+		case <-time.After(500 * time.Millisecond):
```

Comment:
Any special reason for not using a ticker here? Using a `time.After` instead will put more strain on the GC because it creates a new `time.Timer` on each call to `select`. Furthermore, the timer won't be GC:ed until it expires which means that there might be a fair amount of them in memory on a busy handler.

Reply:
So I was worried about drift between the ticker and the loop. At the end of the day, what we want is to say "if after X milliseconds there's nothing in the channel, see about stopping the loop". But the ticker is ticking whether or not we're in the select. So if, for example, we receive a message and spend 99ms processing it, then when the loop comes around we'll only wait 1ms before checking if we're closing.

I guess I could use a ticker and just use `Reset` right before the select, actually, I'll change it to do that